tests/test_models.py

```diff
@@ -719,7 +718,6 @@ class TestNTNHighMemory(_BaseNTNTest):
 
     model_kwargs = {
         'num_slices': 2,
-        'automatic_memory_optimization': False,
     }
```

Comment:
Now, `TestNTNLowMemory` = `TestNTNHighMemory`?

Reply:
Fixed in 364d9e35ad024cd1b040c6029ff8edf105914070.

Reply:
don't forget to explicitly set the evaluator's automatic_memory_optimization

Reply:
Now i'm wondering if/how these should get passed through to the CLI, which tests the whole pipeline

Reply:
I solved this in b973e7ce, but now I'm wondering if it makes sense to store this as a boolean in instances of the `Evaluator` class, or rather to just make it an argument to the `Evaluator.evaluate()` function. This is the last thing I'd like to discuss before finishing/merging this PR (besides passing CI ;))

Reply:
I was also thinking about it, and in the initial commits, we defined `amo` as a function argument of Evaluator.evaluate(), and TrainingLoop.train(). But when defining the `EarlyStopper`, we need to pass the information to the Evaluator, too, so that we might need to define `amo` as an attribute of the `EarlyStopper`. Therefore, I prefer to define it directly as an attribute of the `Evaluator`/`TrainingLoop`

Reply:
I think it is something which you either want to have for the whole training / evaluation or not; so it won't change for multiple calls of the evaluate method, which is why storing it as state of the evaluator seems fine for me.

Reply:
Okay then everything is good as it stands